hugegraph-api/src/main/java/com/baidu/hugegraph/api/graph/EdgeAPI.java

```diff
@@ -157,6 +159,68 @@ public String create(@Context GraphManager manager,
         });
     }
 
+    /**
+     * TODO: Adapter for createIfNotExist or del it.
+     * Batch update steps are same like vertices
+     **/
+    @PUT
+    @Timed
+    @Decompress
+    @Path("batch")
+    @Consumes(APPLICATION_JSON)
+    @Produces(APPLICATION_JSON_WITH_CHARSET)
+    public String update(@Context HugeConfig config,
+                         @Context GraphManager manager,
+                         @PathParam("graph") String graph,
+                         EdgeRequest req) {
+        LOG.debug("Graph [{}] update edges: {}", graph, req.jsonEdges);
+        checkUpdatingBody(req.jsonEdges);
+
+        HugeGraph g = graph(manager, graph);
+        checkBatchSize(config, req.jsonEdges);
+
+        Map<String, JsonEdge> maps = new HashMap<>(req.jsonEdges.size());
+        TriFunction<HugeGraph, Object, String, Vertex> getVertex =
+                    req.checkVertex ? EdgeAPI::getVertex : EdgeAPI::newVertex;
+
+        return this.commit(config, g, maps.size(), () -> {
+
+             // 1.Put all newEdges' properties into map (combine first)
+            req.jsonEdges.forEach(newEdge -> {
+                Id labelId = g.edgeLabel(newEdge.label).id();
+                String id = newEdge.id != null ? newEdge.id :
+                            this.getEdgeId(g, labelId, newEdge);
+                this.updateExistElement(newEdge, maps.get(id),
+                                        req.updateStrategies);
+                maps.put(id, newEdge);
+            });
+
+            // 2.Get all oldEdges and update with new ones
+            List<String> ids = new ArrayList<>(maps.size());
+            maps.keySet().forEach(key -> ids.add(key));
+            Iterator<Edge> oldEdges = g.edges(ids.toArray());
+            oldEdges.forEachRemaining(oldEdge -> {
+                updateExistElement(oldEdge,
+                                   maps.get(oldEdge.id().toString()),
```

Comment:
first I want to use a common method called `getElementId` to get EdgeId & VertexId together, so I use `String` for both
now the methods are separated. maybe use `Map<Object, JsonVertex>` &` Map<String, JsonEdge>` is more simple because JsonVertex & JsonEdge 's  `id ` property's type is `Object` & `String`.
if use Id for them, both of them need to convert again, but seems unnecessary?

Reply:
first I want to use a common method called `getElementId` to get EdgeId & VertexId together, so I use String for both
now the methods are separated. maybe use `Map<Object, JsonVertex>` & `Map<String, JsonEdge> `is more simple because JsonVertex & JsonEdge 's `id` property's type is` Object `& `String`.
if use `Id` type for them, both of them need to convert again, but seems unnecessary?

Reply:
both Edge Id and Vertex Id are instance of Id, EdgeId is a subclass of id.